### PR TITLE
Ml on gcp training at scale

### PR DESCRIPTION
**Pull request by BenoitDherin**
The first notebook 0_export_data_from_bq_to_gcs.ipynb is only meant to export the training data to GCP. The updated lab notebook is 1_training_at_scale.ipynb

### CODE REVIEW

File: courses/machine_learning/production_ml_systems/taxifare/trainer/task.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import model
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--batch_size",
+        help = "Batch size for training steps",
```
Comment (munnm) [inline courses/machine_learning/production_ml_systems/taxifare/trainer/task.py]: I guess no extra space around the "=" here and throughout.

File: courses/machine_learning/production_ml_systems/taxifare/setup.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""Setup."""
+from os import path
+from setuptools import find_packages
+from setuptools import setup
+
+HERE = path.abspath(path.dirname(__file__))
+
+with open(path.join(HERE, 'requirements.txt')) as f:
+  requirements = []
```
Comment (munnm) [inline courses/machine_learning/production_ml_systems/taxifare/setup.py]: indentation here

File: courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb [hunk 3]
```diff
@@ -0,0 +1,228 @@
+{
```
Comment (munnm) [inline courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb]: <p>looks good</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (munnm) [inline courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb]: <p>Would it make sense to do something like this to create the dataset to handle errors (if it already exists) or perhaps too cumbersome?</p><p> </p><p> </p><p>from google.cloud import bigquery</p><p>bq = bigquery.Client(project = PROJECT)</p><p> </p><p>dataset = bigquery.Dataset(bq.dataset("taxifare"))</p><p>try:</p><p>&nbsp;&nbsp;bq.create_dataset(dataset) # will fail if dataset already exists</p><p>&nbsp;&nbsp;print("Dataset created")</p><p>except:</p><p>&nbsp;&nbsp;print("Dataset already exists")</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb [hunk 4]
```diff
@@ -0,0 +1,701 @@
+{
```
Comment (munnm) [inline courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb]: <p>typos: notebo -&gt; notebook and CCS -&gt; GCS</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (munnm) [inline courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb]: <p>When I ran this I got an error initially and had to run <code>gcloud auth configure-docker</code> to get it to work.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

File: courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb (hunk 3 above)
Comment (BenoitDherin) [reply courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb]: <p>Done!</p>

File: courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb (hunk 4 above)
Comment (BenoitDherin) [reply courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb]: <p>Added! Thanks for spotting that!</p>

File: courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
```
Comment (lakshmanok) [inline courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb]: <p>shouldn;'t need this on AI Platform</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

File: courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb [hunk 6]
```diff
@@ -0,0 +1,702 @@
...
```
Comment (lakshmanok) [inline courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb]: <p>You are installing the module and then running just task.py. I would suggest using python -m to run it as a module.  This way:</p><p> </p><p><code>FROM gcr.io/deeplearning-platform-release/tf2-cpu</code></p><p> </p><p><code>#RUN python3 -m pip install --upgrade --quiet tf-nightly-2.0-preview</code></p><p><code>RUN python3 -m pip install --upgrade --quiet cloudml-hypertune</code></p><p> </p><p><code>COPY . /flights</code></p><p><code>ENV PYTHONPATH ${PYTHONPATH}:/flights</code></p><p><code>ENTRYPOINT ["python3", "-m", "trainer.task"]</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/584/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

File: courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb (hunk 5 above)
Comment (BenoitDherin) [reply courses/machine_learning/production_ml_systems/0_export_data_from_bq_to_gcs.ipynb]: <p>Done!</p>

File: courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb (hunk 6 above)
Comment (BenoitDherin) [reply courses/machine_learning/production_ml_systems/1_training_at_scale.ipynb]: <p>Done!</p>

**Comment by BenoitDherin** [comment]
Resolved the comments in the pipeline PR

**Pull request closed without merging**